core/sdk-core/src/main/java/software/amazon/awssdk/core/batchmanager/BatchManager.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.batchmanager;
+
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.internal.batchmanager.DefaultBatchManager;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+
+// TODO: Add javadoc comments. Should I move JavaDoc comments from DefaultBatchManager here? Or keep comments in both files?
```

Comment:
We should move the Javadocs from the implementation to the interface

Reply:
Moved javadocs from implementation to interface (kept them for implementation variables).